big-int/src/bigint.cpp

```diff
@@ -0,0 +1,33 @@
+#include "bigint.hpp"
+#include <cmath>
+#include <regex>
+
+namespace libbig
+{
+largeInt::largeInt() : sign(true), number("0")
+{
+}
+
+largeInt::largeInt(int x) : sign(x >= 0), number(std::to_string(std::abs(x)))
+{
+}
+
+largeInt::largeInt(int64_t x) : sign(x >= 0), number(std::to_string(std::abs(x)))
+{
+}
+
+largeInt::largeInt(long long x) : sign(x >= 0), number(std::to_string(std::abs(x)))
+{
+}
+
+largeInt::largeInt(std::string x)
+{
+    if (!std::regex_match(x, std::regex("(\\-)?(\\d)+")))
```

Comment:
this regex expression does not consider letters 'a-z' && 'A-Z' and other symbols. in order words only number 0-9 should be allowed, if input is other than 0-9 through an exception `invalid integer: integer must contains digits from 0-9`

Reply:
The regex matches any string which may or may not start with a "-", followed by at least one digit. Nothing else will go through. The function `std::regex_match` does a full string match, not a search for that pattern.

If you want, I could add tests for that regex pattern too.